xds/internal/clusterspecifier/rls/rls_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package rls
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	_ "google.golang.org/grpc/balancer/rls"
+	"google.golang.org/grpc/internal/grpctest"
+	"google.golang.org/grpc/internal/proto/grpc_lookup_v1"
+	"google.golang.org/grpc/internal/testutils"
+	_ "google.golang.org/grpc/xds/internal/balancer/cdsbalancer"
+	"google.golang.org/grpc/xds/internal/clusterspecifier"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+// TestParseClusterSpecifierConfig tests the parsing functionality of the RLS
+// Cluster Specifier Plugin.
+func (s) TestParseClusterSpecifierConfig(t *testing.T) {
+	tests := []struct {
+		name    string
+		rlcs    proto.Message
+		wantErr bool
+	}{
+		{
+			name:    "invalid-rls-cluster-specifier",
+			rlcs:    rlsClusterSpecifierConfigError,
+			wantErr: true,
+		},
+		{
+			name: "valid-rls-cluster-specifier",
+			rlcs: rlsClusterSpecifierConfigWithoutTransformations,
+		},
+	}
+	for _, test := range tests {
+		cs := clusterspecifier.Get("type.googleapis.com/grpc.lookup.v1.RouteLookupClusterSpecifier")
+		if cs == nil {
+			t.Fatal("Error getting cluster specifier")
+		}
+		_, err := cs.ParseClusterSpecifierConfig(test.rlcs)
```

Comment:
Validate the result?  (I'm assuming it would be marshaling the result to JSON and compare to a string.)

Reply:
I was expecting to see something like this here now:

```go
lbCfgJSON, err := json.Marshal(lbCfg)
// handle err
var got interface{}
err := json.Unmarshal(lbCfgJSON, &got)
// handle err
wantJSON := `
{
	// JSON that we wanted
}
`
var want interface{}
err := json.Unmarshal(wantJSON, &want)
// handle err
if diff := cmp.Diff(got, want); diff != "" {
	// error
}
```

Reply:
Done. Although I kept the want as a struct, and then Marshaled and then Unmarshaled. I think it's cleaner.